### DISCUSSION TITLE
Document the required outbound Connections for isolated instances

### DISCUSSION THREAD

**Discussion by tweak19** [nextcloud/all-in-one#5424] [Ideas] [answered]
Hi,
I wanna ask if anybody can tell me what DNS-Destinations are required to completly isolate my nextcloud aio setup.
My idea is to block it completly from the internet (outbound) and only allow the needed destinations.

Thanks for your feedback

**Comment by szaimen** [comment]
Hi, see https://github.com/nextcloud/all-in-one?tab=readme-ov-file#can-i-run-aio-offline-or-in-an-airgapped-system

**Comment by tweak19** [comment]
Hi,
yeah that's not new. But as far as I know nextcloud is able to run airgapped.
So i think it should be possible to filter only for the needed dns-destinations, like market-place, dockerhub, e.g.

Nobody done this before?

**Reply by szaimen** [reply, marked as answer]
> Nobody done this before?

It looks like you are the first one. So you would need to investigate which domains are actually required to be reachable yourself. We could then add this to the readme. 

Some starting points: 
- https://docs.nextcloud.com/server/stable/admin_manual/installation/harden_server.html#connections-to-remote-servers
- https://github.com/nextcloud/all-in-one/blob/97dcab58d8f4eeb437a5790587d0de9c8a02ee0b/php/src/Docker/DockerHubManager.php#L30
- https://github.com/nextcloud/all-in-one/blob/97dcab58d8f4eeb437a5790587d0de9c8a02ee0b/php/src/Docker/DockerHubManager.php#L38
- probably more
